chrome/content/coniks/coniks.js

```diff
@@ -0,0 +1,119 @@
+this.EXPORTED_SYMBOLS = ["coniks"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource:///modules/imServices.jsm");
+Cu.import("chrome://otr/content/coniks/context.js");
+Cu.import("chrome://otr/content/helpers.js");
+Cu.import("chrome://otr/content/otr.js");
+
+function toId(account, protocol) {
+  return `${account}@${protocol}`;
+}
+
+function accToId(acc) {
+  return toId(acc.normalizedName, acc.protocol.normalizedName);
+}
+
+var coniks = {
+  isEnable: false,
+
+  init: function() {
+    coniks.setPrefs();
+    coniks.isEnabled = coniks.prefs.getBoolPref("enabled");
+    if (!coniks.isEnabled)
+      return;
+
+    this._contexts = new Map();
+    for (let acc of helpers.getAccounts()) {
+      try {
+        coniks._contexts.set(accToId(acc), new AccountContext({
+          account: acc.normalizedName,
+          protocol: acc.protocol.normalizedName
+        }));
+      } catch(e) { Cu.reportError(e); }
+    }
+    return coniks.loadFiles();
+  },
+
+  destroy: function() {
+    if (!coniks.isEnabled)
+      return;
+  },
+
+  onAccountCreated: function(acc) {
+    let account = acc.normalizedName;
+    let protocol = acc.protocol.normalizedName;
+    let fingerprint = otr.privateKeyFingerprint(account, protocol);
+    try {
```

Comment:
I'd rather a pattern like,

```
onAccountCreated: function(acc) {
  return Promise.resolve().then(function() {
    ... throwable business
  });
} 
```

The thenable will catch the throw and reject the promise, which will bubble up to the caller to handle the error.  But I can make this change.  I think we've approached the end of my suggestions on this patch.  Getting ready to merge.